pandas/core/indexes/base.py

```diff
@@ -4881,6 +4881,9 @@ def _ensure_index(index_like, copy=False):
     if hasattr(index_like, 'name'):
         return Index(index_like, name=index_like.name, copy=copy)
 
+    if is_iterator(index_like):
```

Comment:
I *think* this covers generators as well?

Reply:
sure